File: javascript/pedido.js
const RepositorioPedido = require ('./repositorioPedido');
const ServicoPedido = require('./servicoPedido');


// Criando uma instância do RepositorioPedido
const repositorio = new RepositorioPedido();

// Injetando a dependência no ServicoPedido
const  servico = new ServicoPedido(repositorio);

// Simulando um pedido
const pedido = {id: 1, descricao: "Pedido exemplo"}

// Processando o pedido usando o serviço
servico.processarPedido(pedido);

File: dependency_injection_java/Pedido.java
package dependency_injection_java;

public class Pedido {
    private int id;
    private String descricao;


    public Pedido(int id, String descricao) {
        this.id = id;
        this.descricao = descricao;
    }

    @Override
    public String toString() {
        return "Pedido [id= " + id + ", descricao= " + descricao + "]";
    }

}